docs/build-as-dll.md

```diff
@@ -0,0 +1,11 @@
+# Design of building OpenTelemetry C++ SDK and exporters as DLL
+
+**NOTE**: the current build produces a single DLL which includes API, SDK and
+exporters. This **MAY** change in future.
+
+OpenTelemetry C++ can be built into single DLL named opentelemetry_cpp.dll on
+Windows, with CMake flag `OPENTELEMETRY_BUILD_DLL` set to `ON`.
+
+To instrument an application based on the DLL, please define macro
+`OPENTELEMETRY_BUILD_IMPORT_DLL` before including any OpenTelemetry API header
+files in the application source files.
```

Comment:
Do you this we should update the documentation here too - https://github.com/open-telemetry/opentelemetry-cpp/blob/main/INSTALL.md#building-shared-libs-for-windows ?

Reply:
Thanks for the catching. Updated https://github.com/open-telemetry/opentelemetry-cpp/blob/main/INSTALL.md#building-shared-libs-for-windows.